stream.go

```diff
@@ -177,24 +179,19 @@ NewStream:
 				stream.Events <- ev
 			case <-stream.closer:
 				if r != nil {
-					r.Close()
+					_ = r.Close()
 					// allow the decoding goroutine to terminate
-					for {
-						if _, ok := <-errs; !ok {
-							break
-						}
+					for range errs {
 					}
-					for {
-						if _, ok := <-events; !ok {
-							break
-						}
+					for range events {
 					}
 				}
 				break NewStream
 			case <-retryChan:
 				var err error
 				r, err = stream.connect()
 				if err != nil {
+					r = nil
```

Comment:
Better safe than sorry here.  I find relying on interfaces returning a nil value when they return an error to be fraught with peril.